### PR TITLE
Setup releases and update mechanism for desktop app

### PR DESCRIPTION
**Pull request by ashu8912**

### CODE REVIEW

File: app/electron/main.js [hunk 1]
```diff
@@ -133,8 +140,34 @@ function createWindow () {
...
   });
+  mainWindow.once('ready-to-show', () => {
+    autoUpdater.checkForUpdatesAndNotify();
+  })
+}
+
+autoUpdater.autoDownload = true
+
+function sendStatusToWindow(text) {
+  log.info(text);
```
Comment (joaquimrocha) [inline app/electron/main.js:151]: Please add a context message here unless the log does it automatically:
```
log.info(`Sending status to window: ${text}`);
```

File: app/electron/main.js [hunk 2]
```diff
@@ -133,8 +140,34 @@ function createWindow () {
...
+  mainWindow.webContents.send('message', text);
 }
 
+autoUpdater.on('update-not-available', (info) => {
+  sendStatusToWindow('Update not available.');
```
Comment (joaquimrocha) [inline app/electron/main.js:156]: You're using a period here, but not in the other messages. Please remove it from here.

File: app/electron/main.js [hunk 3]
```diff
@@ -133,8 +140,34 @@ function createWindow () {
...
+})
+
+autoUpdater.on('update-available', () => {
+  sendStatusToWindow('Update is available')
+});
+
+autoUpdater.on('update-downloaded', () => {
+  sendStatusToWindow('Update downloaded')
+});
+
```
Comment (joaquimrocha) [inline app/electron/main.js:170]: The `package-lock.json` change belongs to the previous patch.

**Review by joaquimrocha — CHANGES_REQUESTED** [review]

File: app/package.json [hunk 4]
```diff
@@ -9,14 +9,23 @@
     "copy-server-binary": "cd ../ && make backend && cp -r ./backend/server ./app/electron/",
     "start": "npm run copy-server-binary && cd ../ && make run-backend  & cd ../frontend/ && export BROWSER=none && npm start",
     "build": "npm run copy-server-binary && cd ../frontend/ && export PUBLIC_URL='.' npm run build && cp -r ../app/electron/. build/electron",
-    "package": "mkdir -p build && cp -r ../frontend/build/* ./build && electron-builder build --mac --win --linux -c.extraMetadata.main=build/electron/main.js --publish never",
-    "serve-dev": "export ELECTRON_START_URL=http://localhost:3000 && electron ."
+    "package": "mkdir -p build && cp -r ../frontend/build/* ./build && electron-builder build --linux -c.extraMetadata.main=build/electron/main.js --publish never",
```
[the remark this answers is not in the document]
Comment (ashu8912) [reply app/package.json]: true, its default to what system we are building it

**Review by joaquimrocha — APPROVED** [review]

**Pull request merged**